redbot/cogs/audio/apis/spotify.py

```diff
@@ -151,12 +152,18 @@ async def get_access_token(self) -> Optional[str]:
 
     async def post(
         self, url: str, payload: MutableMapping, headers: MutableMapping = None
-    ) -> MutableMapping:
+    ) -> Optional[MutableMapping]:
         """Make a POST call to spotify."""
         async with self.session.post(url, data=payload, headers=headers) as r:
-            data = await r.json(loads=json.loads)
-            if r.status != 200:
```

Comment:
We still should be checking for status code as some (if not most) non-200 responses contain valid JSON. Yes, with current usage this doesn't technically matter but we should be doing it nonetheless.
If we do this, we could probably even simplify `get_access_token()` and remove the `try-except` with `KeyError` from it and just directly assign instead.